spatialmedia/gui.py

```diff
@@ -39,73 +40,232 @@
+            self.var_spherical.set(0)
+            self.var_3d.set(0)
+
+        self.update_state()
+
+    def action_inject_delay(self):
+        stereo = None
+        if (self.var_3d.get()):
+            stereo = "top-bottom"
+
+        xml = spherical.generate_spherical_xml(stereo=stereo)
+
+        console = Console()
+        spherical.inject_metadata(
+            self.in_file, self.save_file, xml, console.append)
+        self.set_message("Successfully saved file to %s\n"
+                         % ntpath.basename(self.save_file))
+        self.button_open.configure(state="normal")
+        self.button_quit.configure(state="normal")
+        self.update_state()
+
+    def action_inject(self):
         """Inject metadata into a new save file."""
-        self.save_file = tkFileDialog.asksaveasfilename(**self.open_options)
-        if self.save_file:
-            self.set_text("Saved file to: %s\n" % self.save_file)
-            xml = spherical.generate_spherical_xml()
-            spherical.inject_metadata(self.in_file, self.save_file, xml, self.append_text)
-
-    def set_text(self, text):
-        """Updates text box contents."""
-        self.textbox.config(state=NORMAL)
-        self.textbox.delete(1.0, END)
-        self.textbox.insert(END, text)
-        self.textbox.config(state=DISABLED)
-
-    def append_text(self, text):
-        self.textbox.config(state=NORMAL)
-        self.textbox.insert(END, "\n")
-        self.textbox.insert(END, text)
-        self.textbox.config(state=DISABLED)
+        baseFileName = os.path.splitext(ntpath.basename(self.in_file))[0]
+        self.save_options["initialfile"] = baseFileName + "_injected.mp4"
+        self.save_file = tkFileDialog.asksaveasfilename(**self.save_options)
+        if not self.save_file:
+            return
+
+        self.set_message("Saving file to %s" % ntpath.basename(self.save_file))
+
+        self.disable_state()
+        self.master.after(100, self.action_inject_delay)
+
+    def action_set_spherical(self):
+        self.update_state()
+
+    def action_set_3d(self):
+        self.update_state()
+
+    def enable_state(self):
+        self.button_open.configure(state="normal")
+        self.button_quit.configure(state="normal")
+
+    def disable_state(self):
+        self.checkbox_spherical.configure(state="disabled")
+        self.checkbox_3D.configure(state="disabled")
+        self.options_projection.configure(state="disabled")
+        self.button_inject.configure(state="disabled")
+        self.button_open.configure(state="disabled")
+        self.button_quit.configure(state="disabled")
+
+    def update_state(self):
+        self.checkbox_spherical.configure(state="normal")
+        if self.var_spherical.get():
+            self.checkbox_3D.configure(state="normal")
+            self.options_projection.configure(state="normal")
+            self.button_inject.configure(state="normal")
+        else:
+            self.checkbox_3D.configure(state="disabled")
```

Comment:
What happens if a user:
1) Opens a mp4 that already has spherical metadata (UI is updated as such)
2) User unchecks the "Spherical" checkbox
3) User clicks "Save As"?

I would expect the metadata to be -not present- in the new file. However, based on action_inject_delayed() it would seem that metadata is injected. These also seems to be true if the original file doesn't have metadata (Since var_spherical is never checked prior to injecting).